### PR TITLE
[datadog_service_account] Add service account resource

### PR DESCRIPTION
**Pull request by mnguyendatadog**
Adds the service_account resource, which is a subclass of the User resource.

addresses #1266

### CODE REVIEW

File: datadog/resource_datadog_service_account.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceDatadogServiceAccount() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a Datadog service account resource. This can be used to create and manage Datadog service accounts.",
+		CreateContext: resourceDatadogServiceAccountCreate,
+		ReadContext:   resourceDatadogServiceAccountRead,
+		UpdateContext: resourceDatadogServiceAccountUpdate,
+		DeleteContext: resourceDatadogUserDelete,
```
Comment (skarimo) [inline datadog/resource_datadog_service_account.go:20]:
```suggestion
		DeleteContext: resourceDatadogServiceAccountDelete,
```

File: datadog/tests/resource_datadog_service_account_test.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+				Config: fmt.Sprintf(`
+        data "datadog_role" "ro_role" {
+          filter = "Datadog Read Only Role"
+        }
+
+        resource "datadog_service_account" "some_test_updated_service_account" {
+          email = "some_linked_users_email@test.com"
+          name = "Service account linked to some user"
+          roles = [data.datadog_role.ro_role.id]
+        }`),
```
Comment (skarimo) [inline datadog/tests/resource_datadog_service_account_test.go:67]: Nit: Could you move this out to a separate func/variable.

File: datadog/tests/resource_datadog_service_account_test.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+				Config: fmt.Sprintf(`
+        data "datadog_role" "ro_role" {
+          filter = "Datadog Read Only Role"
+        }
+
+        resource "datadog_service_account" "some_test_updated_service_account" {
+          email = "some_linked_users_email@test.com"
+          name = "New name for the service account"
+          roles = []
+        }`),
```
Comment (skarimo) [inline datadog/tests/resource_datadog_service_account_test.go:89]: Same as above

File: datadog/resource_datadog_service_account.go [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
```
[the remark this answers is not in the document]
Comment (mnguyendatadog) [reply datadog/resource_datadog_service_account.go:25]: @skarimo what spec are you referring to?

File: datadog/tests/resource_datadog_service_account_test.go (hunk 2 above)
Comment (mnguyendatadog) [reply datadog/tests/resource_datadog_service_account_test.go:67]: I'll move

File: datadog/tests/resource_datadog_service_account_test.go (hunk 3 above)
Comment (mnguyendatadog) [reply datadog/tests/resource_datadog_service_account_test.go:89]: Same as above

File: datadog/resource_datadog_service_account.go (hunk 1 above)
Comment (mnguyendatadog) [reply datadog/resource_datadog_service_account.go:20]: sorry this was leftover code - there is no functional difference between deleting a user resource versus a service account resource, so it should be `resourceDatadogUserDelete`

File: datadog/resource_datadog_service_account.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (nkzou) [reply datadog/resource_datadog_service_account.go:25]: The spec at https://github.com/DataDog/datadog-api-spec/blob/master/spec/v2/users.yaml and/or https://github.com/DataDog/datadog-api-spec/blob/master/spec/v2/service_accounts.yaml both have a title field defined.

[the remark this answers is not in the document]
Comment (mnguyendatadog) [reply datadog/resource_datadog_service_account.go:25]: adding

**Comment by jorhett** [comment]
@mnguyendatadog the only change required to get past the test failure is shown in the test output at https://github.com/DataDog/terraform-provider-datadog/actions/runs/3840441200/jobs/6539504078#step:11:178

It's a one-character change to get this moving forward.

**Comment by jorhett** [comment]
@skarimo @nkzou I've tested this in our existing production environment and it works 👍 

@mnguyendatadog small docs patch improvement:

```diff
--- a/docs/resources/service_account.md
+++ b/docs/resources/service_account.md
@@ -42,4 +42,10 @@ resource "datadog_service_account" "bar" {

 - `id` (String) The ID of this resource.

+## Import

+Import is supported using the following syntax:
+
+```shell
+terraform import datadog_service_account.example_sa 6f1b44c0-30b2-11eb-86bc-279f7c1ebaa4
+```
```

File: datadog/resource_datadog_service_account.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (mnguyendatadog) [reply datadog/resource_datadog_service_account.go:25]: @nkzou I don't seem to see `title` used that often in the UX or code for service accounts. It's not even part of the schema for its parent `User`. Are those specs correct/up to date?

**Review by nkzou — APPROVED** [review]

**Pull request merged**